### PR TITLE
Added the mentoring.md file to Pig Latin exercise

### PR DESCRIPTION
**Pull request by thomasantony12**
The mentoring.md file has been added to provide a comprehensive guide for mentors assisting students with the Pig Latin exercise. This document includes clear instructions, detailed explanations of the translation rules, step-by-step implementation guidance, and example code to ensure that mentors can effectively support students in understanding and solving the problem.

### CODE REVIEW

**Comment by IsaacG** [comment]
Mentoring notes should assume the mentor is familiar with the exercise already. The instructions should reside in the instructions, and not be duplicated in the mentoring notes. The mentoring notes should focus on common areas of improvement, not on how the exercise should be completed.

**Review by thomasantony12 — COMMENTED** [review]
check it now

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Pig Latin
+
+
+## Common Areas of Improvement
+When mentoring students on the Pig Latin exercise, it's essential to focus on common areas where they might struggle. Here are some key points to keep in mind:
+
+1. Understanding the Rules Clearly
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:7]: This doesn't have anything to do wit Pig Latin. Mentoring typically occurs after the solution passes the tests. If the tests pass, presumably the solution is implemented correctly and all edge cases have been accounted for.

I would recommend mentoring this exercise a couple of times then reviewing some common themes that come up in the mentoring discussions. I wouldn't expect that mentors need to call out edge cases when the tests already enforce those.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    - Rule Comprehension: Ensure that students fully understand each of the Pig Latin translation rules before diving into coding. Misunderstanding the rules often leads to incorrect implementations.
+    - Edge Cases: Highlight the importance of edge cases such as words starting with "xr" or "yt", and those containing "qu".
+2. String Manipulation Skills
+    - Splitting and Joining Strings: Students should be comfortable using functions to split sentences into words and join words back into sentences.
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:11]: If the solution passes tests, this is presumably not something that the mentor needs to bring up.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - String Slicing: Emphasize the importance of correctly slicing strings to rearrange parts of the word as per the rules.
+    - Prefix Checking: Encourage using built-in string functions like starts with to simplify rule checks.
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:13]: s/starts with/`str.startswith()`/

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+3. Iterating Over Strings
+    - Looping Over Characters: Ensure students understand how to iterate over characters in a string, especially when finding the first vowel or handling specific patterns.
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:15]: Regex solutions don't necessarily involve this. This sounds like a description of an implementation, not a common talking point.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    - Range-based Loops: Encourage using range-based loops or list comprehensions for readability and efficiency.
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:16]: Range-based loops are generally not the best approach for most iteration.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+4. Condition Handling
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:17]: Is there any specific examples in Pig Latin where this applies? The second point sounds pretty generic and not specific to this exercise.

**Comment by thomasantony12** [comment]
Based on your suggestions, I have redone the file. check the changes and let me know if any other updates are needed. Thank you.

File: tracks/python/exercises/armstrong-numbers/mentoring.md [hunk 7]
```diff
@@ -37,8 +37,8 @@ def is_armstrong_number(number):
     # With a generator expression passed to sum
     total = sum(digit ** count for digit in digits)
 ```
-Students are often unaware of [generator expressions][pep-289] and that they can be passed directly to functions that accept iterables.
-The first line uses a comprehension to build a `list`  for `sum()` to iterate through.
+Students are often unaware of [generator expressions][pep-289] and that they can be passed directly to functions that accept iteration.
```
Comment (IsaacG) [inline tracks/python/exercises/armstrong-numbers/mentoring.md:40]: Is this intentionally part of this PR?

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+The problem requires the student to rearrange the English word based on the given rules in the instruction.
+
+## Reasonable solutions
+
+This exercise can be solved by using one `for` loop.
+
+```
+def translate(text):
+    return ' '.join([pig(word) for word in text.split(' ')])
+def pig(word):
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:14]: Functions should be separated by two blank lines.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    if word[:2] in ['yt', 'xr']:
+        return word + "ay"
+    if word[:2] in ['qu']:
+        return word[2:] + word[:2] + "ay"
+    if set('aieou') - set(word) == set('aieou') and 'yt' not in word:
+        return word[1:] + word[:1] + "ay"
+    return pig(word[1:] + word[0])
+```
+
+Exercise solved using two `for` loops.
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:28]: Generally mentoring notes have a single reasonable solution. Is there something significant about the second solution that warrants two solutions? The number of `for` statements isn't very descriptive of what's special about the second solution.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+            return word[i:] + word[:i] + "ay"
+        elif char == 'y' and i > 0:
+            # Rule 4: Consonants followed by "y"
+            return word[i:] + word[:i] + "ay"
+    
+    # If no vowels are found, just add "ay" to the word (edge case)
+    return word + "ay"
+```
+
+Prefer to use the first approach. Using multiple `for` loops in your code can sometimes lead to readability issues and inefficiencies. Remember that readability and maintainability are crucial.
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:60]: Exercism markdown should have one sentence per line.

Mentoring notes shouldn't be telling mentors what solution students should use. There are a lot of approaches which are valid. There is no one correct approach. 

Are there common talking points you've returned to when mentoring this exercise? That's what mentoring notes should contain.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    my_list = ["apple", "banana", "cherry"]
+    separator = ", "
+    result = separator.join(my_list)
+    print(result)
+    # Output: "apple, banana, cherry"
+    ```
+
+2. Also students may unfamiliar with `set()`:
+
+    The `set()` function is a powerful data structure in Python.
```
Comment (IsaacG) [inline tracks/python/exercises/pig-latin/pig-latin/mentoring.md:76]: Mentors should be familiar with `set`. There is no need to explain what `set()` is to mentors. This point says "consider `set`" but doesn't explain why set is useful here or how it helps solve this exercise.

**Review by IsaacG — CHANGES_REQUESTED** [review]

File: tracks/python/exercises/armstrong-numbers/mentoring.md (hunk 7 above)
Comment (thomasantony12) [reply tracks/python/exercises/armstrong-numbers/mentoring.md:40]: Oh sorry about that. I committed to the main branch. it was a mistake.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md (hunk 8 above)
Comment (thomasantony12) [reply tracks/python/exercises/pig-latin/pig-latin/mentoring.md:14]: okay.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md (hunk 9 above)
Comment (thomasantony12) [reply tracks/python/exercises/pig-latin/pig-latin/mentoring.md:28]: okay, I will remove the 2nd solution.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md (hunk 10 above)
Comment (thomasantony12) [reply tracks/python/exercises/pig-latin/pig-latin/mentoring.md:60]: Okay.

File: tracks/python/exercises/pig-latin/pig-latin/mentoring.md (hunk 11 above)
Comment (thomasantony12) [reply tracks/python/exercises/pig-latin/pig-latin/mentoring.md:76]: Okay, I will be adding that

**Comment by thomasantony12** [comment]
I will create a new pull request for the changes I made. I am closing this PR because I accidentally added 2 other files which are not for this PR.
